### ISSUE TITLE
add disk parameter "sparse" to template file/sharedfile

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#315]
Originally reported of Google Code with ID 259.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".
# gnt-cluster --version
gnt-cluster (ganeti v2.6.0) 2.6.0
# gnt-cluster version
Software version: 2.6.0
Internode protocol: 2060000
Configuration format: 2060000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>
# cat /etc/SuSE-release
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 2

creating disks with template file/sharedfile results in sparse files.

# gnt-instance modify --disk add:size=100G testvm.loc
Fri Aug 10 11:14:15 2012 * disk 1, size 100.0G
Modified instance testvm.loc
 - disk/1 -> add:size=102400,mode=rw

# ls -alh /srv/ganeti/shared-file-storage/testvm.loc/disk1
-rwx------ 1 nobody nobody 100G Aug 10 11:14 /srv/ganeti/shared-file-storage/testvm.loc/disk1
# du -sh /srv/ganeti/shared-file-storage/testvm.loc/disk1
0       /srv/ganeti/shared-file-storage/testvm.loc/disk1

This is an extrem kind of thin provisioning (storage overcommit) and therefor not good for production. It would be nice if disk template file/sharedfile has a boolean parameter "sparse", which is default "false" and creates disks non-sparse.

TIA, Sascha.
```
Originally added on 2012-08-10 09:19:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
"gnt-cluster modify --prealloc-wipe-disks true" does partialy what I want. maybe some hints in the docs (gnt-cluster(8) and gnt-instance(8)) is enough. this method is good for adding instances' disks. but grow-disk seems not to respect the wipe settings?
```
Originally added on 2012-08-13 07:23:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Note that the prealloc-wipe-disks now works for grow-disk as well (implemented and will be released in 2.7). Is this enough for you?
```
Originally added on 2012-10-30 11:26:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thanks, of course 2.7 is enough.
```
Originally added on 2012-10-30 19:34:10 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I was more asking if the prealloc being fixed is enough or if you actually think we should have a sparse flag. I'll take it is enough, and mark this fixed; feel free to reopen if I misunderstood :)
```
Originally added on 2012-10-30 22:21:18 +0000 UTC.

Added Labels: Type-Defect
Changed State: Fixed
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
